File: math/armstrong_number/C#/ArmstrongNumber.cs
using System;

/// <summary>
/// Algorithm to tell if a given number is Armstrong or not
/// </summary>
public class ArmstrongNumber 
{
	public static void main(string[] args)
	{
		var number = Console.ReadLine();
		if (IsArmstrongNumber(number))
		{
			Console.WriteLine($"Number {number} is Armstrong");
		} 
		else 
		{
			Console.WriteLine($"Number {number} is not Armstrong");
		}
	}

	public static bool IsArmstrongNumber(string number)
	{
		ulong sum = 0;
		foreach (var digit in number)
		{
			var num = Convert.ToUInt64(digit) - 48;
			sum += (ulong)Math.Pow(num, number.Length);
		}

		return sum.ToString().Equals(number);
	}
}

File: math/armstrong_number/dart/armstrong_number.dart
import 'dart:math';

void main() {
  isArmstrongNumber(371);
  isArmstrongNumber(810);
  isArmstrongNumber(1634);
  isArmstrongNumber(2020);
}

void isArmstrongNumber(int number) {
  int sum = 0, digits = 0;

  for (int i = number; i != 0; i ~/= 10) digits++;
  for (int i = number; i != 0; i ~/= 10) {
    sum += pow(i % 10, digits);
  }

  if (sum == number)
    print("${number} is a armstrong number.");
  else
    print("${number} is not a armstrong number.");
}
